flake.nix

```diff
@@ -9,91 +9,167 @@
+            environment.systemPackages = [ (mkIf cfg.keyb0xx.enable (pkgs.keyb0xx.override { keyb0xxconfig = cfg.keyb0xx.config; })) ];
           };
         };
-      };
-      config = {
-        nixpkgs.overlays = [ (mkIf cfg.overlay.enable self.overlay) ];
-        services.udev.extraRules = mkIf cfg.gcc.rules.enable cfg.gcc.rules.rules;
-        boot.kernelModules = mkIf cfg.gcc.oc-kmod.enable ["gcadapter_oc"];
-        boot.extraModulePackages = mkIf cfg.gcc.oc-kmod.enable [
-          config.boot.kernelPackages.gcadapter-oc-kmod
-        ];
-        nix = mkIf cfg.cache.enable {
-          settings.substituters = [ "https://ssbm-nix.cachix.org" ];
-          settings.trusted-public-keys = [ "ssbm-nix.cachix.org-1:YN104LKAWaKQIecOphkftXgXlYZVK/IRHM1UD7WAIew=" ];
+      homeManagerModule = { pkgs, config, ... }:
+        let
+          cfg = config.ssbm;
+        in
+        with nixpkgs.lib; {
+          options.ssbm = {
+            slippi-launcher = {
+              enable = mkEnableOption "Install Slippi Launcher";
+              # Game settings
+              isoPath = mkOption {
+                default = "";
+                type = types.str;
+                description = "The path to an NTSC Melee ISO.";
+              };
+              launchMeleeOnPlay = mkEnableOption "Launch Melee in Dolphin when the Play button is pressed." // { default = true; };
+              enableJukebox = mkEnableOption "Enable in-game music via Slippi Jukebox. Incompatible with WASAPI." // { default = true; };
+              # Replay settings
+              rootSlpPath = mkOption {
+                default = "${config.home.homeDirectory}/Slippi";
+                type = types.str;
+                description = "The folder where your SLP replays should be saved.";
+              };
+              useMonthlySubfolders = mkEnableOption "Save replays to monthly subfolders";
+              spectateSlpPath = mkOption {
+                default = "${cfg.slippi-launcher.rootSlpPath}/Spectate";
+                type = types.nullOr types.str;
+                description = "The folder where spectated games should be saved.";
+              };
+              extraSlpPaths = mkOption {
+                default = [ ];
+                type = types.listOf types.str;
+                description = "Choose any additional SLP directories that should show up in the replay browser.";
+              };
+              # Netplay
+              netplayDolphinPath = mkOption {
+                default = "${pkgs.slippi-netplay}";
+                type = types.str;
+                description = "The path to the folder containing the Netplay Dolphin Executable";
+              };
+              # Playback
+              playbackDolphinPath = mkOption {
+                default = "${pkgs.slippi-playback}";
+                type = types.str;
+                description = "The path to the folder containing the Playback Dolphin Executable";
+              };
+            };
```

Comment:
really cool - thanks for taking the time to document these options and to expose them in the module!

Reply:
It was my pleasure!